brd-android/trade/src/main/java/com/fabriik/trade/ui/features/swap/SwapInputViewModel.kt

```diff
@@ -0,0 +1,118 @@
+package com.fabriik.trade.ui.features.swap
+
+import android.app.Application
+import androidx.lifecycle.viewModelScope
+import com.fabriik.common.ui.base.FabriikViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import java.math.BigDecimal
+
+class SwapInputViewModel(
+    application: Application
+) : FabriikViewModel<SwapInputContract.State, SwapInputContract.Event, SwapInputContract.Effect>(
+    application
+) {
+
+    init {
+        refreshQuote()
+    }
+
+    override fun createInitialState() = SwapInputContract.State(
+        timer = QUOTE_TIMER,
+        quoteLoading = true,
+        originCurrency = "BSV",
+        originCurrencyBalance = BigDecimal.TEN,
+        destinationCurrency = "BTC",
+        rateOriginToDestinationCurrency = BigDecimal.ONE
+    )
+
+    override fun handleEvent(event: SwapInputContract.Event) {
+        when (event) {
+            SwapInputContract.Event.DismissClicked ->
+                setEffect { SwapInputContract.Effect.Dismiss }
+
```

Comment:
Are empty lines after each case intentional?

Reply:
Yes, I added empty lines to improve readability